server/src/main/java/com/defold/extender/Extender.java

```diff
@@ -124,14 +126,66 @@ private class ProGuardContext {
 
     private static final boolean DM_DEBUG_DISABLE_PROGUARD = System.getenv("DM_DEBUG_DISABLE_PROGUARD") != null;
 
-    Extender(String platform, File sdk, File jobDirectory, File uploadDirectory, File buildDirectory, Map<String,String> env) throws IOException, ExtenderException {
-        this.jobDirectory = jobDirectory;
-        this.uploadDirectory = uploadDirectory;
-        this.buildDirectory = buildDirectory;
+    static public class Builder {
+        private String platform;
+        private File sdk;
+        private File jobDirectory;
+        private File buildDirectory;
+        private File uploadDirectory;
+        private Map<String, String> env = new HashMap<String, String>();
+        private MetricsWriter metricsWriter;
+
+        public Builder() { }
+
+        public Builder setPlatform(String platform) {
+            this.platform = platform;
+            return this;
+        }
+
+        public Builder setSdk(File sdk) {
+            this.sdk = sdk;
+            return this;
+        }
+
+        public Builder setJobDirectory(File jobDirectory) {
+            this.jobDirectory = jobDirectory;
+            return this;
+        }
+
+        public Builder setBuildDirectory(File buildDirectory) {
+            this.buildDirectory = buildDirectory;
+            return this;
+        }
+
+        public Builder setUploadDirectory(File uploadDirectory) {
+            this.uploadDirectory = uploadDirectory;
+            return this;
+        }
+
+        public Builder setEnv(Map<String, String> env) {
+            this.env = env;
+            return this;
+        }
+
+        public Builder setMetricsWriter(MetricsWriter writer) {
+            this.metricsWriter = writer;
+            return this;
+        }
+
+        public Extender build() throws IOException, ExtenderException {
+            return new Extender(this);
+        }
+    };
+
+    private Extender(Builder builder) throws IOException, ExtenderException {
+        this.jobDirectory = builder.jobDirectory;
+        this.uploadDirectory = builder.uploadDirectory;
+        this.buildDirectory = builder.buildDirectory;
+        this.metricsWriter = builder.metricsWriter;
```

Comment:
Add metricsWriter reference to use it inside build methods to count builds by target type.